### DISCUSSION TITLE
Support for c++ modules for gcc compiler

### DISCUSSION THREAD

**Discussion by shubham4060** [ccache/ccache#1099] [Development]
Hi,

Currently the ccache does support c++ modules but only for clang compiler and in direct+depend mode. So I wanted to ask if and when the modules support will be added for gcc compiler and if it will be, will preprocessor mode be supported for c++ modules with gcc compiler?

**Comment by jrosdahl** [comment]
Hi,

There is no specific plan to support C++ modules, but contributions in the area are welcome. I have not personally studied GCC's implementation, but I would be surprised if it could be made to work without an enabled depend mode.
